.github/workflows/integrational-tests.yml

```diff
@@ -52,41 +63,73 @@ jobs:
 
     - name: Wait for deployment to finish
       run: |-
-        kubectl wait --for=condition=ready pod/hazelcast-0 --timeout=120s
-        kubectl wait --for=condition=ready pod/hazelcast-1 --timeout=120s
-        kubectl wait --for=condition=ready pod/hazelcast-2 --timeout=120s
+        kubectl wait --for=condition=ready pod/hz-hazelcast-0 --timeout=120s
+        kubectl wait --for=condition=ready pod/hz-hazelcast-1 --timeout=120s
+        kubectl wait --for=condition=ready pod/hz-hazelcast-2 --timeout=120s
 
-    - name: Creating LoadBalancer for each pods
+    - name: Wait for external IP to get assigned
       run: |-
-        for pod in $(kubectl get pods -o jsonpath="{.items[*].metadata.name}"); do \
-          kubectl create service loadbalancer ${pod} --tcp=5701 -o yaml --dry-run | kubectl set selector --local -f - "statefulset.kubernetes.io/pod-name=${pod}" -o yaml | kubectl create -f -; \
-          kubectl label service ${pod} test=true; \
-        done;
+        EXTERNAL_IP=$(kubectl get svc hz-hazelcast --output="jsonpath={.status.loadBalancer.ingress[0].ip}")
+        while [ "$EXTERNAL_IP" == "" ]; do
+          sleep 10
+          EXTERNAL_IP=$(kubectl get svc hz-hazelcast --output="jsonpath={.status.loadBalancer.ingress[0].ip}")
+        done
 
-    - name: Fetch k8s master IP
-      id: master-ip
-      run: |-
-        ip=$(kubectl cluster-info | grep master | awk -Fat '{gsub(/ /, "", $0); print $NF}')
-        echo "masterIp=$ip" >> $GITHUB_ENV
+        echo "EXTERNAL_IP=${EXTERNAL_IP}" >> $GITHUB_ENV
 
-    - name: Run Maven tests
+    - name: Test Java Client
       run: |-
-        secret=$(kubectl get secret -o name | grep hazelcast)
-        apiToken=$(kubectl get $secret -o jsonpath={.data.token} | base64 --decode | xargs echo)
-        caCertificate=$(kubectl get $secret -o jsonpath='{.data.ca\.crt}' | base64 --decode)
-
-        KUBERNETES_MASTER="$masterIp" API_TOKEN=$apiToken CA_CERTIFICATE=$caCertificate mvn test -q
-      timeout-minutes: 10
+        EXTERNAL_IP="${{ env.EXTERNAL_IP }}"
+        cd java
+        sed -i "s/<EXTERNAL-IP>/$EXTERNAL_IP/g" src/main/java/com/hazelcast/Main.java
+        mvn package
+        java -jar target/*jar-with-dependencies*.jar >> output-java.txt &
+        PID=$!
+        sleep 10
+        kill $PID
+
+        cat output-java.txt | grep 'Successful connection!' -q
+
+    - name: Test NodeJS Client
+      run: |-
+        EXTERNAL_IP="${{ env.EXTERNAL_IP }}"
+        cd nodejs
+        sed -i "s/<EXTERNAL-IP>/$EXTERNAL_IP/g" client.js
+        npm install
+        npm start >> output-nodejs.txt &
+        PID=$!
+        sleep 10
+        kill $PID
+
+        cat output-nodejs.txt | grep 'Successful connection!' -q
+
+    - name: Test Go Client
+      run: |-
+        EXTERNAL_IP="${{ env.EXTERNAL_IP }}"
+        cd go
+        sed -i "s/<EXTERNAL-IP>/$EXTERNAL_IP/g" main.go
+        go get github.com/hazelcast/hazelcast-go-client@v1.0.0-preview.4
```

Comment:
We shouldn't need a separate `go get` step, since the Hz client dependency is in `go.mod`.